### ISSUE TITLE
magit-blame might decode non-ASCII strings incorrectly

### ISSUE THREAD

**Issue by Eli-Zaretskii** [magit/magit#2738]
When the default process-coding-system is not UTF-8, non-ASCII characters in the Git log messages will be displayed incorrectly by magit-blame.  This is because magit-blame doesn't bind coding-system-for-read before running the "git blame" command, so it leaves decoding non-ASCII text to whatever defaults are set up out there.
This means in particular that using magit-blame on Windows will almost always produce incorrect display of non-ASCII characters, because UTF-8 is never the default encoding on Windows.

To reproduce, set the locale's encoding to something other than UTF-8 and then invoke magit-blame on Emacs's lisp/version.el file -- you will see the name of Pavel Janík displayed incorrectly.

More generally, magit should pay attention to i18n.logOutputEncoding, which could be different from UTF-8. Latest Emacs versions introduced a user variable for that, vc-git-log-output-coding-system.

**Comment by tarsius** [comment]
This should be fixed now. Thanks for reporting.

**Comment by tarsius** [comment]
Seems this is the case for `git add` too. https://github.com/magit/magit/issues/581#issuecomment-246483262

**Comment by bojanrajkovic** [comment]
Seems like it might be the case for `git status`, and all other commands as well. I'm encountering something that may be similar when trying to run `magit-status` against a branch that has an emoji name (yes, I know, don't use emoji in branch names :).

**Comment by npostavs** [comment]
Pushed an initial draft fix to [np/git-encoding](https://github.com/magit/magit/compare/np/git-encoding), seems to work. Quoting my commit message:
```
All git commands which print file names and similar encoding in utf-8 on
Windows.  The encoding of commit messages is controlled by
i18n.logOutputEncoding, but it's easier to set that to UTF-8 than try to
distinguish between commands that respect and those that don't.

magit-log-output-coding-system should probably be obsoleted/removed.
```

**Comment by tarsius** [comment]
I've merged #2813.
